recipe-client-addon/lib/Heartbeat.jsm

```diff
@@ -315,12 +313,16 @@ this.Heartbeat = class {
       this.chromeWindow.gBrowser.selectedTab = this.chromeWindow.gBrowser.addTab(this.options.postAnswerUrl.toString());
     }
 
-    if (this.surveyEndTimer) {
-      clearTimeout(this.surveyEndTimer);
-      this.surveyEndTimer = null;
-    }
+    this.endTimerIfPresent("surveyEndTimer");
+
+    this.engagementCloseTimer = setTimeout(() => this.close(), NOTIFICATION_TIME);
```

Comment:
The binding thing happened in a patch revision (cf. last point of https://bugzilla.mozilla.org/show_bug.cgi?id=1330954#c2 ) and I forgot to update this - good find! We could indeed just pass this.close directly.